src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -18,8 +21,22 @@
     private static final Logger LOGGER = LogManager.getLogger(MovieSessionDaoImpl.class);
 
     @Override
-    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
-        return null;
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDateTime date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<MovieSession> criteriaQuery = criteriaBuilder
+                    .createQuery(MovieSession.class);
+            Root<MovieSession> sessionRoot = criteriaQuery.from(MovieSession.class);
+            Predicate idPredicate = criteriaBuilder
+                    .equal(sessionRoot.get("movie"), movieId);
+            Predicate datePredicate = criteriaBuilder
+                    .greaterThan(sessionRoot.get("showtime"), date);
```

Comment:
```suggestion
            Predicate datePredicate = criteriaBuilder
                    .greaterThan(sessionRoot.get("showtime"), date.atStartOfDay());
```
таким чином ти зможеш працювати з LocalDate ( відповідно до умови)